lib/chess_web/controllers/api/game_controller.ex

```diff
@@ -25,22 +27,34 @@ defmodule ChessWeb.Api.GameController do
       |> Repo.get!(id)
 
     changeset = Game.changeset(
-      game, %{board: new_board(game.board, move_params)}
+      game, %{
+        board: Board.move_piece(game.board, move_params),
+        turn: Game.change_turn(game.turn)
+      }
     )
 
     case Repo.update(changeset) do
       {:ok, game} ->
+        GameChannel.update_game(game)
+
         conn
-        |> json(Board.transform(game.board))
+        |> json(game_attrs(conn, game))
     end
   end
 
-  defp new_board(board, move_params) do
-    [from_file, from_rank] = move_params["from"]
-    [to_file, to_rank] = move_params["to"]
-
-    {piece, board} = Map.pop(board, "#{from_file},#{from_rank}")
+  defp game_attrs(conn, game) do
+    %{
+      board: Board.transform(game.board),
```

Comment:
What does `Board.tranform` mean? Is it serializing it?

Reply:
The board is stored in the database as a map of pieces indexed by their position on the board but the Javascript renders the board from a 2 dimensional array of squares. This function transforms the board from a list of pieces and their positions to that 2D array.